In [1]:
from IPython.display import HTML
display(HTML("<style>.container { width:97% !important; }</style>"))

In [2]:
import numpy as np
import pandas as pd
from itertools import product
from pandas.tseries.offsets import DateOffset

import matplotlib.pyplot as plt
import matplotlib.dates as md
from matplotlib.dates import MO, TU, WE, TH, FR, SA, SU
from copy import deepcopy
import datetime
import calendar
from typing import List

In [3]:
import warnings
warnings.filterwarnings("ignore")

In [4]:
from sktime.performance_metrics.forecasting import mean_absolute_scaled_error
from sklearn.metrics import mean_absolute_error
import lightgbm as lgb
from lightgbm import Dataset


In [5]:
import sys
sys.path.append('../..')
from utils.analyze import compute_metrics
from utils.metrics import Bias, Wape, Count, Sum, Mape, MSE

### Data Loader

In [6]:
data = pd.read_csv('data/train.csv')
data['dt'] = pd.to_datetime(data['dt'])

In [7]:
# functions

def create_train_df(df: pd.DataFrame, 
                    START_PREDICT_DATE: str,
                    END_PREDICT_DATE: str
                   ):
    
    START_TRAIN_DATE = pd.to_datetime('2016-01-02')
    START_PREDICT_DATE = pd.to_datetime(START_PREDICT_DATE)
    END_PREDICT_DATE = pd.to_datetime(END_PREDICT_DATE)
    
    range_date = pd.date_range(START_TRAIN_DATE, END_PREDICT_DATE, freq='D')
    unique_shop_id = df['id'].unique().tolist()
    
    train_grid = pd.DataFrame(list(product(unique_shop_id,
                                           range_date)),
                              columns = ['id', 'dt']
                             )
    df_train = train_grid.merge(df, how = 'left', on = ['id', 'dt'])
    df_train['sales'] = df_train['target'].copy()
    df_train['target'] = df_train['target'].fillna(0)
    df_train['split'] = np.where(df_train['dt'] >= START_PREDICT_DATE, 'test', 'train')
    df_train.loc[df_train['dt'] >= START_PREDICT_DATE, 'target'] = np.nan 

    return df_train

def create_calendar_vars(df, date_col):
    """
    Create calendar vars
    """

    def week_of_month(dt):
        first_day = dt.replace(day=1)
        dom = dt.day
        adjusted_dom = dom + first_day.weekday()
        return int(np.ceil(adjusted_dom / 7.0))

    df_ = df.copy()
    df_['year'] = df_[date_col].dt.year.astype('int')
    df_['quarter'] = df_[date_col].dt.quarter.astype('int')
    df_['month'] = df_[date_col].dt.month.astype('int')
    df_['dayofweek'] = df_[date_col].dt.dayofweek.astype('int')
    df_['dayofyear'] = df_[date_col].dt.dayofyear.astype('int')
    df_['dayofmonth'] = df_[date_col].dt.day.astype('int')
    df_['week'] = df_[date_col].dt.isocalendar().week.astype('int') 
    df_['week'] = df_['week'].astype('int')
    df_['is_weekend'] = pd.Series(df_[date_col]).apply(lambda x : x.weekday() in [5, 6]).values 
    df_['is_weekend'] = df_['is_weekend'].astype('category')
    df_['week_of_month'] = pd.Series(df_[date_col]).apply(week_of_month).values
    df_['week_of_month'] = df_['week_of_month'].astype('int')
    df_['first_day_month'] = df_['dayofmonth'].apply(lambda x: x == 1).astype('category')
    df_['second_day_month'] = df_['dayofmonth'].apply(lambda x: x == 2).astype('category')
    df_['is_evenweek'] = df_['week'] // 2 == 0 
    df_['is_evenweek'] = df_['is_evenweek'].astype('category')
    df_['week_of_month + week'] = df_['week_of_month'].astype('str') + '-'  + df_['week'].astype('str')
    df_['week_of_month + week'] = df_['week_of_month + week'].astype('category')
    df_['week + quarter'] = df_['week'].astype('str') + '-'  + df_['quarter'].astype('str') 
    df_['week + quarter'] = df_['week + quarter'].astype('category')
    df_['dayofmonth + quarter'] = df_['dayofmonth'].astype('str') + '-'  + df_['quarter'].astype('str')
    df_['dayofmonth + quarter'] = df_['dayofmonth + quarter'].astype('category')
    df_['is_evenweek + dayofweek'] = df_['is_evenweek'].astype('str') + '-'  + df_['dayofweek'].astype('str')
    df_['is_evenweek + dayofweek'] = df_['is_evenweek + dayofweek'].astype('category')
    return df_



def get_start_end_dates(df, n_splits, col_dates='dt'):
    """
    create dates
    """
    max_date = df[col_dates].max()
    
    from_date = max_date - DateOffset(months=n_splits-1)     
    beg_date = datetime.datetime(from_date.year, from_date.month, 1)
    for i in range(n_splits):
        n_days_in_that_month = calendar.monthrange(beg_date.year, beg_date.month)[1]
        end_date = datetime.datetime(beg_date.year, beg_date.month, n_days_in_that_month)
        yield (beg_date, end_date)
    
        beg_date = end_date + datetime.timedelta(days=1)

In [8]:
def create_lags(df, group_columns: List[str], target_column: str, lags_range: list):
    """
    Create lags
    """
    df_ = df.copy()
    for lag in lags_range:
        df_[f'Lags_{lag}'] = df_.groupby(group_columns)[target_column].shift(lag)
        
    return df_


def create_group_means(df, target_column: str):
    """
    Create grouped means
    """
    df_ = df.copy()
    df_['mean_id'] = df_.groupby(['id'])[target_column].transform(lambda x: x.mean())
    df_['mean_year'] = df_.groupby(['year'])[target_column].transform(lambda x: x.mean())
    df_['mean_quarter'] = df_.groupby(['quarter'])[target_column].transform(lambda x: x.mean())
    df_['mean_week'] = df_.groupby(['week'])[target_column].transform(lambda x: x.mean())
    df_['mean_dayofweek'] = df_.groupby(['dayofweek'])[target_column].transform(lambda x: x.mean())
    df_['mean_dayofweek_even_week'] = df_.groupby(['dayofweek', 'is_evenweek'])[target_column].transform(lambda x: x.mean())
    
    df_['mean_dayofweek_even_id'] = df_.groupby(['dayofweek', 'is_evenweek', 'id'])['target'].transform(lambda x: x.mean())
    df_['max_dayofweek_even_id'] = df_.groupby(['dayofweek', 'is_evenweek', 'id'])['target'].transform(lambda x: x.max())
    df_['min_dayofweek_even_id'] = df_.groupby(['dayofweek', 'is_evenweek', 'id'])['target'].transform(lambda x: x.min())
    df_['diff_dayofweek_even_id'] = df_.groupby(['dayofweek', 'is_evenweek', 'id'])['target'].transform(lambda x: x.max() - x.min())
    
    df_['mean_dayofweek_week_of_month_id'] = df_.groupby(['dayofweek', 'week_of_month', 'id'])['target'].transform(lambda x: x.mean())
    df_['max_dayofweek_week_of_month_id'] = df_.groupby(['dayofweek', 'week_of_month', 'id'])['target'].transform(lambda x: x.max())
    df_['min_dayofweek_week_of_month_id'] = df_.groupby(['dayofweek', 'week_of_month', 'id'])['target'].transform(lambda x: x.min())
    df_['diff_dayofweek_week_of_month_id'] = df_.groupby(['dayofweek', 'week_of_month', 'id'])['target'].transform(lambda x: x.max() - x.min())

                                                   
    return df_


def create_rollings_means(df, target_column: str):
    """
    Create rolling means
    """
    
    df_ = df.copy()
    df_['rolling_mean_id'] = df_.groupby(['id'])[target_column].transform(
        lambda s: s.shift(HORIZON).rolling(window=7, min_periods=1, center=False).agg('mean'))
    df_['rolling_mean_id_28'] = df_.groupby(['id'])[target_column].transform(
        lambda s: s.shift(HORIZON).rolling(window=28, min_periods=1, center=False).agg('mean'))
        
    return df_


def create_seasonal_vars(df: pd.DataFrame, group_columns: List[str], target_column=str):
    """
    Create seasonal features
    """
    train = df[df['split'] == 'train']
    train['trend'] = train.groupby(group_columns)[target_column].transform(
        lambda s: s.shift(HORIZON).rolling(window=500, min_periods=30, center=True).agg('mean')
    )
    train['trend'] = train['trend'].fillna(method='bfill')
    train["detrended"] = train[target_column] - train["trend"]
    
    coef = train.groupby(group_columns, as_index = False)[['detrended']].mean()
    coef = coef.rename(columns = {'detrended': 'seasonal'})
    
    # Клипануть сезонные коэффициенты
    
    return coef



In [9]:
def filter_dataset(df):
    mask_na = (df['dt'] >= pd.to_datetime('2016-07-01'))\
                & (df['dt'] < pd.to_datetime('2017-01-01'))\
                & (df['id'].isin([4, 8, 10, 13, 17])
                  )
    early_mask = (df['dt'] <= pd.to_datetime('2016-03-01')) 
    mask = mask_na + early_mask
    
    return df[~mask]

In [10]:
def get_valid_index(train: 
                    pd.DataFrame, 
                    valid_size: int = 0.1,
                    random_seed = 0
                   ):
    """
    get valid index and drop them form train
    """
    np.random.seed(random_seed)
    valid_size = np.ceil(len(train) * valid_size).astype(int)
    valid_idx = np.random.choice(train.index, valid_size, replace = False)
    valid = train.loc[valid_idx]
    
    train.drop(valid_idx, axis = 0, inplace = True)
    
    return valid
    

In [11]:
def prepare_datesets(df, features, valid_size, random_seed):
    """
    Prepare dataset for lgbm
    
    """
    
    train = df[(df['split'] == 'train')]
    test = df[df['split'] == 'test']
    valid = get_valid_index(train,
                           valid_size,
                           random_seed,
                       )
    
    cat_columns = train[features].select_dtypes(include='category').columns.tolist()
    
    # lgbm datasets
    train_ds = lgb.Dataset(train[features],
                       train['target'],
                       categorical_feature=cat_columns,
                       )
    
    val_ds = lgb.Dataset(valid[features],
                         valid['target'],
                         categorical_feature=cat_columns,
                         reference=train_ds
                        )
    
    return train_ds, val_ds



In [12]:
def get_predictions(df, model):
    df['prediction'] = model.predict(df[features2train])
    df['prediction'] = np.clip(df['prediction'], a_min=0, a_max = None)
    
    # post_processsing
    claster = [ 2, 3, 4, 5, 6, 7, 8, 9, 10, 13, 14, 16, 17, 18, 19]
    claster_holidays = [pd.to_datetime('2017-06-04')]
    all_holidays = [pd.to_datetime('2017-05-01'),
                    pd.to_datetime('2017-05-14'),
                    pd.to_datetime('2017-05-25'),
                    pd.to_datetime('2017-04-03'),
                    pd.to_datetime('2017-04-06'),
                   ]

    for shop in claster:
        df.loc[(df['id'] == shop)
              & (df['dt'].isin(claster_holidays)), 
               'prediction'
              ] = 0

    df.loc[(df['dt'].isin(all_holidays)), 
           'prediction'
          ] = 0    
        
    return df

In [13]:
def get_metrics(df):
    metrics = {}

    shops_list = df['id'].unique().tolist()
    for shop in shops_list:
        shop_df = df[df['id'] == shop] 
        y_test = shop_df[shop_df['split'] == 'test']['sales'].fillna(0).copy()
        y_pred = shop_df[shop_df['split'] == 'test']['prediction'].copy()
        y_train = shop_df[shop_df['split'] == 'train']['sales'].fillna(0).copy()
        metric =  mean_absolute_scaled_error(y_test, y_pred, y_train = y_train, sp = HORIZON)
        metrics[shop] = metric
        
    return metrics

        
def plot(df: pd.DataFrame, metrics: dict):
    """
    Draw plots with sales & prediction
    """

    df_agg = df.groupby('id')
    fig, axes = plt.subplots(5, 4, figsize = (20, 20))
    axes = axes.ravel()

    for name, group in df_agg:
        group = group[group['split'] == 'test']
        axes[name].plot(group['dt'], group['sales'], label = 'sales',  color = 'red')
        axes[name].plot(group['dt'], group['prediction'], label = 'prediction',  color = 'black')

        metric = metrics[name]
        axes[name].set_title(f'Sales and Prediction\n for Store № {name} -- {metric}')
        axes[name].legend()
        axes[name].grid()
        axes[name].xaxis.set_major_locator(md.WeekdayLocator(MO))
        axes[name].xaxis.set_tick_params(rotation=90)
    plt.tight_layout()
    

### Перекретсная проверка

In [14]:
data = pd.read_csv('data/train.csv')
data['dt'] = pd.to_datetime(data['dt'])

In [15]:
# define features for training
features2train = ['id', 
                  'year', 'quarter', 'month',
                  'dayofmonth', 
                  'dayofweek', 
                  'week', 'is_weekend',
                  'first_day_month', 'second_day_month',
                  'week_of_month + week',
                  'mean_id',
                  'mean_year',
                  'mean_quarter',
                  'mean_week', 
                  'mean_dayofweek',
                  'mean_dayofweek_even_week',
                  'rolling_mean_id',
                  'rolling_mean_id_28',
                  'mean_dayofweek_even_id',
                  'max_dayofweek_even_id', 
                  'min_dayofweek_even_id',
                  'diff_dayofweek_even_id',     
                  'mean_dayofweek_week_of_month_id',
                  'max_dayofweek_week_of_month_id',
                  'min_dayofweek_week_of_month_id',
                  'diff_dayofweek_week_of_month_id',
                  'Lags_91', 
                  'Lags_29', 
                  'Lags_62', 
                  'Lags_122',
                  'seasonal',
                 ]

# boosting hyperparameters
params = {
    'objective': 'mae', 
    'eta': 0.01, # 0.1
    'max_depth' : -1,
    'seed' : 42, 
    'verbose' : -1,
     'num_threads': 8,
    
}


# Best value (rmse): 0.17825
# 	Best params:
# 		n_estimators: 2000
# 		learning_rate: 0.0032824665592938616
            

# Constants
VALID_SIZE = 0.2
VISUALIZE = False

In [16]:
dates = (get_start_end_dates(data, n_splits=4, col_dates='dt'))

for date in dates:
    # define prediction date
    test_start_date = date[0]
    test_end_date = date[1]
    HORIZON = calendar.monthrange(test_start_date.year, test_start_date.month)[1]
    print(test_start_date, test_end_date, HORIZON)
    
    # create dataframe
    df = create_train_df(data, 
                         test_start_date,
                         test_end_date
                        )
    
    # create features
    df = create_calendar_vars(df, date_col='dt')
    
    # create lags
    
    df = create_lags(df, group_columns=['id'], target_column='target', 
                     lags_range = [28, 29, 30, 31, 62, 91, 122, 365])
    
    # gropued means
    df = create_group_means(df, target_column='target')
    
    # rolling means
    df = create_rollings_means(df, target_column='target')
    
    # seasonal coef
    seasona_coef = create_seasonal_vars(df, 
                                        group_columns=['dayofweek', 'is_evenweek', 'id'], 
                                        target_column='target')
    df = df.merge(seasona_coef, on = ['dayofweek', 'is_evenweek', 'id'], how = 'left')
    
    # filter dataset
    df = filter_dataset(df)
    
    # prepeare lgbm dataset
    train_ds, val_ds = prepare_datesets(df, features2train, valid_size=VALID_SIZE, random_seed=0)
    
    # train
    model = lgb.train(params,
                      train_ds,
                      num_boost_round = 1000,
                      valid_sets=[train_ds, val_ds],
                      verbose_eval=500,
                      evals_result=None,
                      )
    
    # prediction
    df = get_predictions(df, model)
    
    # get metrics
    metrics = get_metrics(df)
    print(f'Mean Smape {np.array(list(metrics.values())).mean()}')
    
    if VISUALIZE:
        plot(df, metrics)
    

2017-03-01 00:00:00 2017-03-31 00:00:00 31
[500]	training's l1: 48.9999	valid_1's l1: 53.4864
[1000]	training's l1: 37.9841	valid_1's l1: 48.3325
Mean Smape 0.15609563912108065
2017-04-01 00:00:00 2017-04-30 00:00:00 30
[500]	training's l1: 47.3794	valid_1's l1: 57.3907
[1000]	training's l1: 37.8847	valid_1's l1: 50.9831
Mean Smape 0.2429270911980609
2017-05-01 00:00:00 2017-05-31 00:00:00 31
[500]	training's l1: 50.9441	valid_1's l1: 59.6672
[1000]	training's l1: 41.7753	valid_1's l1: 54.7217
Mean Smape 0.19389037377165524
2017-06-01 00:00:00 2017-06-30 00:00:00 30
[500]	training's l1: 53.4316	valid_1's l1: 59.2664
[1000]	training's l1: 42.174	valid_1's l1: 53.0589
Mean Smape 0.15045403374672306


#### Baseline
Mean Smape 0.15609563912108065 \
Mean Smape 0.2429270911980609 \
Mean Smape 0.19389037377165524 \
Mean Smape 0.15045403374672306 \

Mean Smape 0.1585581324852377 \
Mean Smape 0.22812083356894738 \
Mean Smape 0.19650267164915525 \
Mean Smape 0.15459359524156494

### Подбор гиперпараметров модели

In [40]:
import optuna

In [61]:
def objective(trial, df, n_splits = 4):
    
    all_metric = []
    dates = (get_start_end_dates(data, n_splits=n_splits, col_dates='dt'))

    for date in dates:
        # define prediction date
        test_start_date = date[0]
        test_end_date = date[1]
        HORIZON = calendar.monthrange(test_start_date.year, test_start_date.month)[1]
        print(test_start_date, test_end_date, HORIZON)

        # create dataframe
        df = create_train_df(data, 
                             test_start_date,
                             test_end_date
                            )

        # create features
        df = create_calendar_vars(df, date_col='dt')

        # create lags

        df = create_lags(df, group_columns=['id'], target_column='target', 
                         lags_range = [28, 29, 30, 31, 62, 91, 122, 365])

        # gropued means
        df = create_group_means(df, target_column='target')

        # rolling means
        df = create_rollings_means(df, target_column='target')

        # seasonal coef
        seasona_coef = create_seasonal_vars(df, 
                                            group_columns=['dayofweek', 'is_evenweek', 'id'], 
                                            target_column='target')
        df = df.merge(seasona_coef, on = ['dayofweek', 'is_evenweek', 'id'], how = 'left')

        # filter dataset
        df = filter_dataset(df)

        # prepeare lgbm dataset
        train_ds, val_ds = prepare_datesets(df, features2train, valid_size=VALID_SIZE, random_seed=0)
        
        params = {
            'objective': 'mae', 
            'seed' : 42, 
            'verbose' : -1,
            'num_threads': 8,
            'early_stopping_rounds':100,
            "n_estimators": trial.suggest_categorical("n_estimators", [1000, 2000, 3000, 5000, 7000, 10000]), # n_estimators: 2000
            "learning_rate": trial.suggest_float("learning_rate", 0.001, 0.15) # learning_rate: 0.059863266105050765
            #"num_leaves": trial.suggest_int("num_leaves", 12, 120, step=4), # 16
            #"max_depth": trial.suggest_int("max_depth", 1, 6), # 5
            #"lambda_l1": trial.suggest_int("lambda_l1", 0, 10, step=1), # 0
            #"lambda_l2": trial.suggest_int("lambda_l2", 0, 10, step=1), # 8
            #"bagging_fraction": trial.suggest_float("bagging_fraction", 0.5, 0.95, step=0.1), # 0.9
            #"feature_fraction": trial.suggest_float("feature_fraction", 0.5, 0.95, step=0.1),# 0.6
        }
            

        # train
        model = lgb.train(params,
                          train_ds,
                          num_boost_round = 1000,
                          valid_sets=[train_ds, val_ds],
                          verbose_eval=500,
                          evals_result=None,
                          )

        # prediction
        df = get_predictions(df, model)

        # get metrics
        metrics = get_metrics(df)
        all_metric.append(np.array(list(metrics.values())).mean())
    
    return np.median(all_metric)
    

In [62]:
study = optuna.create_study(direction="minimize", study_name="LGBM Regressor")
func = lambda trial: objective(trial, df, n_splits = 4)
study.optimize(func, n_trials=50)

[I 2023-08-07 12:18:45,904] A new study created in memory with name: LGBM Regressor


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 25.0896	valid_1's l1: 44.6969
[1000]	training's l1: 20.9504	valid_1's l1: 43.6188
[1500]	training's l1: 18.9679	valid_1's l1: 43.2919
[2000]	training's l1: 17.9426	valid_1's l1: 43.125
[2500]	training's l1: 17.2846	valid_1's l1: 43.0066
Early stopping, best iteration is:
[2616]	training's l1: 17.1506	valid_1's l1: 42.9941
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 24.8416	valid_1's l1: 46.6612
[1000]	training's l1: 20.5382	valid_1's l1: 45.2732
[1500]	training's l1: 18.5066	valid_1's l1: 44.5652
[2000]	training's l1: 17.3674	valid_1's l1: 44.2009
[2500]	training's l1: 16.6166	valid_1's l1: 43.9913
[3000]	training's l1: 16.084	valid_1's l1: 43.8837
[3500]	training's l1: 15.6342	valid_1's l1: 43.768
[4000]	training's l1: 15.3213	valid_1's l1: 43.7091
[4500]	training's l1: 15.0737	va

[I 2023-08-07 12:19:18,080] Trial 0 finished with value: 0.20304118040355867 and parameters: {'n_estimators': 5000, 'learning_rate': 0.07988692918026569}. Best is trial 0 with value: 0.20304118040355867.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 30.0986	valid_1's l1: 45.9897
[1000]	training's l1: 24.0068	valid_1's l1: 43.5969
[1500]	training's l1: 21.4476	valid_1's l1: 43.006
[2000]	training's l1: 20.0014	valid_1's l1: 42.723
[2500]	training's l1: 18.943	valid_1's l1: 42.5057
[3000]	training's l1: 18.2017	valid_1's l1: 42.342
Early stopping, best iteration is:
[3104]	training's l1: 18.0906	valid_1's l1: 42.3181
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 28.1913	valid_1's l1: 46.2039
[1000]	training's l1: 23.4251	valid_1's l1: 44.7987
[1500]	training's l1: 21.3215	valid_1's l1: 44.1688
[2000]	training's l1: 20.1653	valid_1's l1: 43.8706
[2500]	training's l1: 19.2486	valid_1's l1: 43.5878
[3000]	training's l1: 18.4501	valid_1's l1: 43.3115
[3500]	training's l1: 17.8128	valid_1's l1: 43.0059
Early stopping, best iteration is

[I 2023-08-07 12:19:49,399] Trial 1 finished with value: 0.20264348374321506 and parameters: {'n_estimators': 5000, 'learning_rate': 0.047897904114985776}. Best is trial 1 with value: 0.20264348374321506.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 94.8608	valid_1's l1: 91.7831
[1000]	training's l1: 58.8185	valid_1's l1: 61.0282
[1500]	training's l1: 46.9312	valid_1's l1: 52.9926
[2000]	training's l1: 41.84	valid_1's l1: 50.2659
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 41.84	valid_1's l1: 50.2659
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 91.8631	valid_1's l1: 99.2331
[1000]	training's l1: 56.3929	valid_1's l1: 65.2446
[1500]	training's l1: 46.1583	valid_1's l1: 55.9135
[2000]	training's l1: 42.1232	valid_1's l1: 53.1096
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 42.1232	valid_1's l1: 53.1096
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 94.9688	valid_1's l1: 94.295
[1000]	training's l1: 59.

[I 2023-08-07 12:20:11,377] Trial 2 finished with value: 0.17559974890753743 and parameters: {'n_estimators': 2000, 'learning_rate': 0.0035626904331113505}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 20.919	valid_1's l1: 44.0319
[1000]	training's l1: 17.5756	valid_1's l1: 43.3873
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 17.5756	valid_1's l1: 43.3873
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 21.2014	valid_1's l1: 46.5189
[1000]	training's l1: 17.8344	valid_1's l1: 45.7833
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 17.8344	valid_1's l1: 45.7833
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 22.0771	valid_1's l1: 46.5862
[1000]	training's l1: 18.6954	valid_1's l1: 45.7831
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 18.6954	valid_1's l1: 45.7831
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores d

[I 2023-08-07 12:20:23,705] Trial 3 finished with value: 0.20446679181775568 and parameters: {'n_estimators': 1000, 'learning_rate': 0.1445989931342997}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 25.4483	valid_1's l1: 45.0494
[1000]	training's l1: 21.3267	valid_1's l1: 44.2395
[1500]	training's l1: 19.3293	valid_1's l1: 43.7775
[2000]	training's l1: 18.0859	valid_1's l1: 43.4449
[2500]	training's l1: 17.3075	valid_1's l1: 43.2537
[3000]	training's l1: 16.7191	valid_1's l1: 43.0768
[3500]	training's l1: 16.292	valid_1's l1: 42.9839
[4000]	training's l1: 15.9802	valid_1's l1: 42.9459
Early stopping, best iteration is:
[4175]	training's l1: 15.8735	valid_1's l1: 42.9245
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 24.9672	valid_1's l1: 47.1599
[1000]	training's l1: 20.9891	valid_1's l1: 46.1151
[1500]	training's l1: 18.3461	valid_1's l1: 45.3458
[2000]	training's l1: 17.2953	valid_1's l1: 45.0449
[2500]	training's l1: 16.6017	valid_1's l1: 44.8702
[3000]	training's l1: 16.1007	

[I 2023-08-07 12:20:55,774] Trial 4 finished with value: 0.19778689166029056 and parameters: {'n_estimators': 10000, 'learning_rate': 0.08090469937821892}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 23.8832	valid_1's l1: 43.2429
[1000]	training's l1: 19.9284	valid_1's l1: 42.4381
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 19.9284	valid_1's l1: 42.4381
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 25.0683	valid_1's l1: 46.5995
[1000]	training's l1: 20.7973	valid_1's l1: 45.4411
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 20.7973	valid_1's l1: 45.4411
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 28.2635	valid_1's l1: 48.8063
[1000]	training's l1: 23.753	valid_1's l1: 47.0697
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 23.753	valid_1's l1: 47.0697
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores do

[I 2023-08-07 12:21:08,542] Trial 5 finished with value: 0.18001945721662838 and parameters: {'n_estimators': 1000, 'learning_rate': 0.08231772242896958}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 26.3517	valid_1's l1: 44.6508
[1000]	training's l1: 21.7423	valid_1's l1: 43.5965
[1500]	training's l1: 19.5401	valid_1's l1: 42.9478
[2000]	training's l1: 18.1082	valid_1's l1: 42.5394
[2500]	training's l1: 17.1288	valid_1's l1: 42.3357
Early stopping, best iteration is:
[2711]	training's l1: 16.8521	valid_1's l1: 42.2647
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 25.5047	valid_1's l1: 45.9212
[1000]	training's l1: 20.8999	valid_1's l1: 44.5791
[1500]	training's l1: 18.8946	valid_1's l1: 43.9563
[2000]	training's l1: 17.6812	valid_1's l1: 43.5618
[2500]	training's l1: 16.8265	valid_1's l1: 43.353
Early stopping, best iteration is:
[2586]	training's l1: 16.7227	valid_1's l1: 43.326
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 ro

[I 2023-08-07 12:21:36,566] Trial 6 finished with value: 0.19997560252393037 and parameters: {'n_estimators': 7000, 'learning_rate': 0.08140587888813264}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 41.1364	valid_1's l1: 50.1346
[1000]	training's l1: 33.8537	valid_1's l1: 47.3499
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 33.8537	valid_1's l1: 47.3499
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 40.6733	valid_1's l1: 52.5005
[1000]	training's l1: 32.1554	valid_1's l1: 48.1338
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 32.1554	valid_1's l1: 48.1338
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 45.0345	valid_1's l1: 56.1144
[1000]	training's l1: 37.1406	valid_1's l1: 52.4115
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 37.1406	valid_1's l1: 52.4115
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores 

[I 2023-08-07 12:21:49,546] Trial 7 finished with value: 0.17738354021586555 and parameters: {'n_estimators': 1000, 'learning_rate': 0.015564096527369741}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 27.6807	valid_1's l1: 45.8383
[1000]	training's l1: 22.752	valid_1's l1: 44.3007
[1500]	training's l1: 20.8639	valid_1's l1: 43.9835
[2000]	training's l1: 19.5215	valid_1's l1: 43.6767
[2500]	training's l1: 18.7601	valid_1's l1: 43.4676
[3000]	training's l1: 18.0513	valid_1's l1: 43.2954
[3500]	training's l1: 17.5364	valid_1's l1: 43.1913
[4000]	training's l1: 17.1002	valid_1's l1: 43.0954
[4500]	training's l1: 16.6769	valid_1's l1: 42.9774
Early stopping, best iteration is:
[4826]	training's l1: 16.4781	valid_1's l1: 42.921
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 26.896	valid_1's l1: 46.7725
[1000]	training's l1: 22.6424	valid_1's l1: 45.5798
[1500]	training's l1: 20.6261	valid_1's l1: 45.0359
[2000]	training's l1: 19.3882	valid_1's l1: 44.7368
[2500]	training's l1: 18.4147	va

[I 2023-08-07 12:22:32,245] Trial 8 finished with value: 0.2077819313405324 and parameters: {'n_estimators': 7000, 'learning_rate': 0.05709499260681381}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 27.8832	valid_1's l1: 44.5607
[1000]	training's l1: 23.4436	valid_1's l1: 43.6761
[1500]	training's l1: 21.2618	valid_1's l1: 43.2198
[2000]	training's l1: 19.9993	valid_1's l1: 42.9505
[2500]	training's l1: 19.1694	valid_1's l1: 42.8449
[3000]	training's l1: 18.4216	valid_1's l1: 42.7527
[3500]	training's l1: 17.9165	valid_1's l1: 42.6642
Early stopping, best iteration is:
[3623]	training's l1: 17.7803	valid_1's l1: 42.6468
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 28.4501	valid_1's l1: 47.3568
[1000]	training's l1: 23.5241	valid_1's l1: 45.9777
[1500]	training's l1: 21.1444	valid_1's l1: 45.2163
[2000]	training's l1: 19.5715	valid_1's l1: 44.8001
[2500]	training's l1: 18.5954	valid_1's l1: 44.4343
[3000]	training's l1: 17.9907	valid_1's l1: 44.3162
[3500]	training's l1: 17.4919

[I 2023-08-07 12:23:11,070] Trial 9 finished with value: 0.19684283129533983 and parameters: {'n_estimators': 7000, 'learning_rate': 0.04955254928375926}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 167.321	valid_1's l1: 163.109
[1000]	training's l1: 124.44	valid_1's l1: 120.835
[1500]	training's l1: 97.089	valid_1's l1: 93.9789
[2000]	training's l1: 79.9446	valid_1's l1: 78.1823
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 79.9446	valid_1's l1: 78.1823
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 165.608	valid_1's l1: 170.343
[1000]	training's l1: 121.972	valid_1's l1: 128.144
[1500]	training's l1: 94.3817	valid_1's l1: 101.486
[2000]	training's l1: 77.0904	valid_1's l1: 84.7899
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 77.0904	valid_1's l1: 84.7899
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 170.239	valid_1's l1: 162.918
[1000]	training's l1: 

[I 2023-08-07 12:23:32,871] Trial 10 finished with value: 0.2192741581842506 and parameters: {'n_estimators': 2000, 'learning_rate': 0.0011456088724464747}. Best is trial 2 with value: 0.17559974890753743.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 82.9913	valid_1's l1: 80.9682
[1000]	training's l1: 52.4992	valid_1's l1: 56.1624
[1500]	training's l1: 43.4506	valid_1's l1: 50.9654
[2000]	training's l1: 39.5499	valid_1's l1: 49.2743
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 39.5499	valid_1's l1: 49.2743
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 79.5947	valid_1's l1: 87.3043
[1000]	training's l1: 50.111	valid_1's l1: 59.5287
[1500]	training's l1: 42.9505	valid_1's l1: 53.3982
[2000]	training's l1: 38.5798	valid_1's l1: 50.5172
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 38.5798	valid_1's l1: 50.5172
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 82.4463	valid_1's l1: 83.7172
[1000]	training's l1:

[I 2023-08-07 12:23:54,267] Trial 11 finished with value: 0.17413337410137147 and parameters: {'n_estimators': 2000, 'learning_rate': 0.004348183306043599}. Best is trial 11 with value: 0.17413337410137147.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 36.1715	valid_1's l1: 47.9292
[1000]	training's l1: 28.3016	valid_1's l1: 44.4157
[1500]	training's l1: 25.5108	valid_1's l1: 43.8517
[2000]	training's l1: 23.6785	valid_1's l1: 43.4942
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 23.6785	valid_1's l1: 43.4942
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 34.9887	valid_1's l1: 48.6995
[1000]	training's l1: 28.4988	valid_1's l1: 46.3585
[1500]	training's l1: 25.4121	valid_1's l1: 45.3661
[2000]	training's l1: 23.7028	valid_1's l1: 44.8985
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 23.7028	valid_1's l1: 44.8985
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 39.6193	valid_1's l1: 53.339
[1000]	training's l1:

[I 2023-08-07 12:24:14,165] Trial 12 finished with value: 0.1889450834639711 and parameters: {'n_estimators': 2000, 'learning_rate': 0.023446870748770398}. Best is trial 11 with value: 0.17413337410137147.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 34.459	valid_1's l1: 47.5595
[1000]	training's l1: 28.2987	valid_1's l1: 45.6585
[1500]	training's l1: 24.82	valid_1's l1: 44.4095
[2000]	training's l1: 22.782	valid_1's l1: 43.733
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 22.782	valid_1's l1: 43.733
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 33.9852	valid_1's l1: 48.2349
[1000]	training's l1: 27.4965	valid_1's l1: 45.9744
[1500]	training's l1: 24.9761	valid_1's l1: 45.3863
[2000]	training's l1: 23.3515	valid_1's l1: 44.9311
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 23.3515	valid_1's l1: 44.9311
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 37.8029	valid_1's l1: 52.2746
[1000]	training's l1: 30.07

[I 2023-08-07 12:24:33,969] Trial 13 finished with value: 0.1771401711942999 and parameters: {'n_estimators': 2000, 'learning_rate': 0.027582818426508483}. Best is trial 11 with value: 0.17413337410137147.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 21.9827	valid_1's l1: 44.8738
[1000]	training's l1: 18.5029	valid_1's l1: 44.2452
[1500]	training's l1: 17.2517	valid_1's l1: 44.0439
[2000]	training's l1: 16.3665	valid_1's l1: 43.8544
[2500]	training's l1: 15.8412	valid_1's l1: 43.7575
Early stopping, best iteration is:
[2405]	training's l1: 15.9185	valid_1's l1: 43.751
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 22.1851	valid_1's l1: 45.6012
[1000]	training's l1: 18.8406	valid_1's l1: 44.6848
[1500]	training's l1: 17.424	valid_1's l1: 44.4004
[2000]	training's l1: 16.473	valid_1's l1: 44.2008
[2500]	training's l1: 15.794	valid_1's l1: 44.0894
Early stopping, best iteration is:
[2414]	training's l1: 15.9041	valid_1's l1: 44.0821
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 roun

[I 2023-08-07 12:24:56,985] Trial 14 finished with value: 0.20549104445346764 and parameters: {'n_estimators': 3000, 'learning_rate': 0.11572541532162678}. Best is trial 11 with value: 0.17413337410137147.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 127.583	valid_1's l1: 123.916
[1000]	training's l1: 82.4938	valid_1's l1: 80.6945
[1500]	training's l1: 62.1065	valid_1's l1: 63.8861
[2000]	training's l1: 52.4078	valid_1's l1: 56.3131
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 52.4078	valid_1's l1: 56.3131
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 125.266	valid_1's l1: 131.558
[1000]	training's l1: 79.7474	valid_1's l1: 87.6514
[1500]	training's l1: 59.6536	valid_1's l1: 68.5063
[2000]	training's l1: 50.4543	valid_1's l1: 59.9534
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 50.4543	valid_1's l1: 59.9534
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 128.772	valid_1's l1: 124.51
[1000]	training's l1:

[I 2023-08-07 12:25:19,343] Trial 15 finished with value: 0.17323944386353118 and parameters: {'n_estimators': 2000, 'learning_rate': 0.002180278335710119}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 32.483	valid_1's l1: 46.814
[1000]	training's l1: 26.6156	valid_1's l1: 45.0487
[1500]	training's l1: 23.7139	valid_1's l1: 44.1961
[2000]	training's l1: 21.8054	valid_1's l1: 43.6543
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 21.8054	valid_1's l1: 43.6543
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 31.1312	valid_1's l1: 47.6255
[1000]	training's l1: 25.7648	valid_1's l1: 46.0643
[1500]	training's l1: 23.3855	valid_1's l1: 45.385
[2000]	training's l1: 21.6901	valid_1's l1: 44.7836
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 21.6901	valid_1's l1: 44.7836
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 34.58	valid_1's l1: 51.0518
[1000]	training's l1: 28.

[I 2023-08-07 12:25:39,219] Trial 16 finished with value: 0.19151651870005182 and parameters: {'n_estimators': 2000, 'learning_rate': 0.03555826846516376}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 23.6456	valid_1's l1: 44.6665
[1000]	training's l1: 20.0447	valid_1's l1: 44.0132
[1500]	training's l1: 18.1779	valid_1's l1: 43.4436
[2000]	training's l1: 17.1081	valid_1's l1: 43.2056
[2500]	training's l1: 16.465	valid_1's l1: 43.0589
[3000]	training's l1: 15.9548	valid_1's l1: 42.9988
Did not meet early stopping. Best iteration is:
[3000]	training's l1: 15.9548	valid_1's l1: 42.9988
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 24.9862	valid_1's l1: 47.0705
[1000]	training's l1: 20.8193	valid_1's l1: 46.0273
[1500]	training's l1: 19.0736	valid_1's l1: 45.5469
[2000]	training's l1: 17.8612	valid_1's l1: 45.2124
[2500]	training's l1: 16.7986	valid_1's l1: 44.7222
[3000]	training's l1: 16.1575	valid_1's l1: 44.5672
Did not meet early stopping. Best iteration is:
[3000]	training's l1:

[I 2023-08-07 12:26:04,874] Trial 17 finished with value: 0.2185849747896552 and parameters: {'n_estimators': 3000, 'learning_rate': 0.10212951556752939}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 43.2651	valid_1's l1: 50.3485
[1000]	training's l1: 35.3878	valid_1's l1: 47.2292
[1500]	training's l1: 31.8101	valid_1's l1: 46.0868
[2000]	training's l1: 29.0198	valid_1's l1: 45.0216
[2500]	training's l1: 27.3815	valid_1's l1: 44.5682
[3000]	training's l1: 25.5523	valid_1's l1: 43.7136
[3500]	training's l1: 24.5246	valid_1's l1: 43.4303
[4000]	training's l1: 23.759	valid_1's l1: 43.3031
[4500]	training's l1: 23.0449	valid_1's l1: 43.1446
[5000]	training's l1: 22.5538	valid_1's l1: 43.069
Early stopping, best iteration is:
[5344]	training's l1: 22.2205	valid_1's l1: 42.9953
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 42.7142	valid_1's l1: 53.4654
[1000]	training's l1: 33.7489	valid_1's l1: 48.362
[1500]	training's l1: 30.1509	valid_1's l1: 46.9688
[2000]	training's l1: 27.5403	va

[I 2023-08-07 12:27:21,800] Trial 18 finished with value: 0.19098268375626226 and parameters: {'n_estimators': 10000, 'learning_rate': 0.013224666867692221}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 31.7187	valid_1's l1: 46.9212
[1000]	training's l1: 25.8744	valid_1's l1: 44.9788
[1500]	training's l1: 22.822	valid_1's l1: 43.9475
[2000]	training's l1: 21.0453	valid_1's l1: 43.4815
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 21.0453	valid_1's l1: 43.4815
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 30.4747	valid_1's l1: 47.2659
[1000]	training's l1: 24.977	valid_1's l1: 45.4764
[1500]	training's l1: 22.6495	valid_1's l1: 44.6883
[2000]	training's l1: 21.0515	valid_1's l1: 44.2617
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 21.0515	valid_1's l1: 44.2617
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 34.1159	valid_1's l1: 51.3521
[1000]	training's l1: 

[I 2023-08-07 12:27:50,757] Trial 19 finished with value: 0.20095835051195976 and parameters: {'n_estimators': 2000, 'learning_rate': 0.03784286617758959}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 26.6126	valid_1's l1: 44.58
[1000]	training's l1: 22.2621	valid_1's l1: 43.6181
[1500]	training's l1: 19.9341	valid_1's l1: 43.0076
[2000]	training's l1: 18.6519	valid_1's l1: 42.8038
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 18.6519	valid_1's l1: 42.8038
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 26.3538	valid_1's l1: 46.0411
[1000]	training's l1: 21.9459	valid_1's l1: 44.9928
[1500]	training's l1: 20.2604	valid_1's l1: 44.6768
[2000]	training's l1: 18.8473	valid_1's l1: 44.2152
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 18.8473	valid_1's l1: 44.2152
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 30.7065	valid_1's l1: 51.1654
[1000]	training's l1: 

[I 2023-08-07 12:28:20,520] Trial 20 finished with value: 0.20210236552980268 and parameters: {'n_estimators': 2000, 'learning_rate': 0.062351522513193924}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 130.298	valid_1's l1: 126.555
[1000]	training's l1: 84.9679	valid_1's l1: 82.4901
[1500]	training's l1: 64.185	valid_1's l1: 65.2147
[2000]	training's l1: 53.3542	valid_1's l1: 56.6423
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 53.3542	valid_1's l1: 56.6423
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 128.188	valid_1's l1: 134.5
[1000]	training's l1: 81.9568	valid_1's l1: 89.6719
[1500]	training's l1: 61.6013	valid_1's l1: 70.5621
[2000]	training's l1: 51.5466	valid_1's l1: 61.0073
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 51.5466	valid_1's l1: 61.0073
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 131.294	valid_1's l1: 126.786
[1000]	training's l1: 8

[I 2023-08-07 12:28:54,510] Trial 21 finished with value: 0.17844998145260205 and parameters: {'n_estimators': 2000, 'learning_rate': 0.0020931584830814662}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 40.2801	valid_1's l1: 49.1367
[1000]	training's l1: 33.3897	valid_1's l1: 46.5006
[1500]	training's l1: 29.0679	valid_1's l1: 44.6035
[2000]	training's l1: 26.7363	valid_1's l1: 43.9598
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 26.7363	valid_1's l1: 43.9598
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 41.0872	valid_1's l1: 52.7668
[1000]	training's l1: 32.7593	valid_1's l1: 48.3104
[1500]	training's l1: 27.8059	valid_1's l1: 46.2494
[2000]	training's l1: 25.6485	valid_1's l1: 45.4844
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 25.6485	valid_1's l1: 45.4844
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 44.7724	valid_1's l1: 56.3794
[1000]	training's l1

[I 2023-08-07 12:29:22,446] Trial 22 finished with value: 0.17415771112756467 and parameters: {'n_estimators': 2000, 'learning_rate': 0.016061675086170978}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 38.8249	valid_1's l1: 48.8147
[1000]	training's l1: 32.2046	valid_1's l1: 46.4617
[1500]	training's l1: 28.9017	valid_1's l1: 45.3123
[2000]	training's l1: 26.6122	valid_1's l1: 44.6409
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 26.6122	valid_1's l1: 44.6409
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 38.358	valid_1's l1: 51.2947
[1000]	training's l1: 30.7825	valid_1's l1: 47.7674
[1500]	training's l1: 27.4739	valid_1's l1: 46.6257
[2000]	training's l1: 25.725	valid_1's l1: 46.1746
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 25.725	valid_1's l1: 46.1746
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 43.0517	valid_1's l1: 55.3749
[1000]	training's l1: 3

[I 2023-08-07 12:29:52,173] Trial 23 finished with value: 0.17849835610363296 and parameters: {'n_estimators': 2000, 'learning_rate': 0.018244204801371438}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 32.1476	valid_1's l1: 45.9571
[1000]	training's l1: 26.5357	valid_1's l1: 44.2895
[1500]	training's l1: 24.0317	valid_1's l1: 43.832
[2000]	training's l1: 22.3377	valid_1's l1: 43.5354
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 22.3377	valid_1's l1: 43.5354
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 31.7632	valid_1's l1: 48.3865
[1000]	training's l1: 26.5027	valid_1's l1: 46.6209
[1500]	training's l1: 23.7349	valid_1's l1: 45.786
[2000]	training's l1: 22.1271	valid_1's l1: 45.349
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 22.1271	valid_1's l1: 45.349
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 34.3977	valid_1's l1: 50.8745
[1000]	training's l1: 27

[I 2023-08-07 12:30:21,059] Trial 24 finished with value: 0.199323894753703 and parameters: {'n_estimators': 2000, 'learning_rate': 0.033466983122606674}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 42.7794	valid_1's l1: 50.8923
[1000]	training's l1: 34.4986	valid_1's l1: 47.5488
[1500]	training's l1: 30.7017	valid_1's l1: 46.2956
[2000]	training's l1: 28.253	valid_1's l1: 45.3914
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 28.253	valid_1's l1: 45.3914
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 41.813	valid_1's l1: 52.3687
[1000]	training's l1: 32.9271	valid_1's l1: 47.4921
[1500]	training's l1: 29.3881	valid_1's l1: 46.4884
[2000]	training's l1: 27.134	valid_1's l1: 45.7295
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 27.134	valid_1's l1: 45.7295
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 45.6437	valid_1's l1: 56.4928
[1000]	training's l1: 36.

[I 2023-08-07 12:30:52,404] Trial 25 finished with value: 0.17695277078281513 and parameters: {'n_estimators': 2000, 'learning_rate': 0.014238253866746646}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 30.5832	valid_1's l1: 45.6135
[1000]	training's l1: 24.4201	valid_1's l1: 43.5551
[1500]	training's l1: 21.8562	valid_1's l1: 42.9751
[2000]	training's l1: 20.4371	valid_1's l1: 42.7772
[2500]	training's l1: 19.3621	valid_1's l1: 42.548
[3000]	training's l1: 18.6534	valid_1's l1: 42.4071
Did not meet early stopping. Best iteration is:
[3000]	training's l1: 18.6534	valid_1's l1: 42.4071
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 29.2895	valid_1's l1: 46.7917
[1000]	training's l1: 24.0721	valid_1's l1: 45.2586
[1500]	training's l1: 21.8977	valid_1's l1: 44.7069
[2000]	training's l1: 20.413	valid_1's l1: 44.2696
[2500]	training's l1: 19.569	valid_1's l1: 44.0754
[3000]	training's l1: 18.6942	valid_1's l1: 43.8444
Did not meet early stopping. Best iteration is:
[3000]	training's l1: 1

[I 2023-08-07 12:31:30,845] Trial 26 finished with value: 0.1971980319885816 and parameters: {'n_estimators': 3000, 'learning_rate': 0.04106920719176935}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 48.0622	valid_1's l1: 53.1752
[1000]	training's l1: 37.7726	valid_1's l1: 48.5011
[1500]	training's l1: 33.8062	valid_1's l1: 46.9981
[2000]	training's l1: 31.3723	valid_1's l1: 46.2438
[2500]	training's l1: 29.0858	valid_1's l1: 45.46
[3000]	training's l1: 27.4591	valid_1's l1: 44.8875
[3500]	training's l1: 26.1487	valid_1's l1: 44.3754
[4000]	training's l1: 25.1289	valid_1's l1: 44.0893
[4500]	training's l1: 24.3073	valid_1's l1: 43.9201
[5000]	training's l1: 23.6832	valid_1's l1: 43.7555
Did not meet early stopping. Best iteration is:
[5000]	training's l1: 23.6832	valid_1's l1: 43.7555
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 46.6278	valid_1's l1: 56.4515
[1000]	training's l1: 36.9039	valid_1's l1: 50.1235
[1500]	training's l1: 31.5647	valid_1's l1: 47.4413
[2000]	training's 

[I 2023-08-07 12:32:32,381] Trial 27 finished with value: 0.18874728165929286 and parameters: {'n_estimators': 5000, 'learning_rate': 0.010443940395144449}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 34.4338	valid_1's l1: 46.721
[1000]	training's l1: 28.3476	valid_1's l1: 44.6062
[1500]	training's l1: 24.8359	valid_1's l1: 43.4236
[2000]	training's l1: 22.9595	valid_1's l1: 43.0278
[2500]	training's l1: 21.622	valid_1's l1: 42.7363
[3000]	training's l1: 20.7612	valid_1's l1: 42.5664
[3500]	training's l1: 20.1541	valid_1's l1: 42.4206
[4000]	training's l1: 19.5878	valid_1's l1: 42.2652
[4500]	training's l1: 19.0686	valid_1's l1: 42.1466
[5000]	training's l1: 18.6801	valid_1's l1: 42.0561
[5500]	training's l1: 18.3301	valid_1's l1: 41.97
[6000]	training's l1: 18.0232	valid_1's l1: 41.8901
[6500]	training's l1: 17.7291	valid_1's l1: 41.7741
[7000]	training's l1: 17.4905	valid_1's l1: 41.7176
[7500]	training's l1: 17.2213	valid_1's l1: 41.6454
[8000]	training's l1: 17.028	valid_1's l1: 41.5976
[8500]	training's l1: 16.8418	valid_1's l1: 41.5112
[9000]	training's

[I 2023-08-07 12:34:00,482] Trial 28 finished with value: 0.1975310864361346 and parameters: {'n_estimators': 10000, 'learning_rate': 0.02840814332382004}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 25.4874	valid_1's l1: 44.2554
[1000]	training's l1: 21.1359	valid_1's l1: 43.4023
[1500]	training's l1: 19.3338	valid_1's l1: 42.9376
[2000]	training's l1: 18.176	valid_1's l1: 42.654
[2500]	training's l1: 17.3568	valid_1's l1: 42.4282
[3000]	training's l1: 16.7707	valid_1's l1: 42.3223
[3500]	training's l1: 16.3468	valid_1's l1: 42.242
[4000]	training's l1: 15.976	valid_1's l1: 42.1713
[4500]	training's l1: 15.6551	valid_1's l1: 42.1171
[5000]	training's l1: 15.3813	valid_1's l1: 42.0758
Did not meet early stopping. Best iteration is:
[5000]	training's l1: 15.3813	valid_1's l1: 42.0758
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 25.8987	valid_1's l1: 46.8791
[1000]	training's l1: 21.982	valid_1's l1: 45.8626
[1500]	training's l1: 20.0699	valid_1's l1: 45.3938
[2000]	training's l1:

[I 2023-08-07 12:34:55,845] Trial 29 finished with value: 0.20010114374920154 and parameters: {'n_estimators': 5000, 'learning_rate': 0.0684510664495426}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 23.9074	valid_1's l1: 43.9663
[1000]	training's l1: 19.9679	valid_1's l1: 43.1652
[1500]	training's l1: 18.231	valid_1's l1: 42.8039
[2000]	training's l1: 17.0909	valid_1's l1: 42.5554
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 17.0909	valid_1's l1: 42.5554
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 23.9386	valid_1's l1: 46.4483
[1000]	training's l1: 20.6049	valid_1's l1: 45.5483
[1500]	training's l1: 18.444	valid_1's l1: 44.9833
[2000]	training's l1: 17.3614	valid_1's l1: 44.7313
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 17.3614	valid_1's l1: 44.7313
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 25.0798	valid_1's l1: 47.3779
[1000]	training's l1: 

[I 2023-08-07 12:35:25,226] Trial 30 finished with value: 0.20521925691629211 and parameters: {'n_estimators': 2000, 'learning_rate': 0.09669336398932449}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 130.957	valid_1's l1: 127.257
[1000]	training's l1: 86.1088	valid_1's l1: 83.5401
[1500]	training's l1: 64.5872	valid_1's l1: 65.6212
[2000]	training's l1: 53.9066	valid_1's l1: 56.9995
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 53.9066	valid_1's l1: 56.9995
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 128.673	valid_1's l1: 134.857
[1000]	training's l1: 82.347	valid_1's l1: 89.9016
[1500]	training's l1: 61.6455	valid_1's l1: 70.4324
[2000]	training's l1: 51.6346	valid_1's l1: 61.0193
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 51.6346	valid_1's l1: 61.0193
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 132.025	valid_1's l1: 127.625
[1000]	training's l1:

[I 2023-08-07 12:35:58,118] Trial 31 finished with value: 0.17946957597135 and parameters: {'n_estimators': 2000, 'learning_rate': 0.0020774278726393785}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 54.7518	valid_1's l1: 57.965
[1000]	training's l1: 40.7037	valid_1's l1: 49.8554
[1500]	training's l1: 36.5474	valid_1's l1: 48.3522
[2000]	training's l1: 34.007	valid_1's l1: 47.5516
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 34.007	valid_1's l1: 47.5516
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 52.1906	valid_1's l1: 61.6038
[1000]	training's l1: 39.8202	valid_1's l1: 51.9257
[1500]	training's l1: 33.6033	valid_1's l1: 48.1771
[2000]	training's l1: 30.8326	valid_1's l1: 47.2495
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 30.8326	valid_1's l1: 47.2495
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 55.6251	valid_1's l1: 62.701
[1000]	training's l1: 44

[I 2023-08-07 12:36:30,293] Trial 32 finished with value: 0.18026003072457397 and parameters: {'n_estimators': 2000, 'learning_rate': 0.008153499266635448}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 37.2123	valid_1's l1: 48.0068
[1000]	training's l1: 31.3057	valid_1's l1: 46.0694
[1500]	training's l1: 26.6584	valid_1's l1: 44.1453
[2000]	training's l1: 24.7204	valid_1's l1: 43.6681
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 24.7204	valid_1's l1: 43.6681
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 35.7109	valid_1's l1: 49.3961
[1000]	training's l1: 29.6622	valid_1's l1: 47.0172
[1500]	training's l1: 26.2592	valid_1's l1: 45.9582
[2000]	training's l1: 24.0756	valid_1's l1: 45.0574
Did not meet early stopping. Best iteration is:
[2000]	training's l1: 24.0756	valid_1's l1: 45.0574
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 40.8367	valid_1's l1: 54.2663
[1000]	training's l1

[I 2023-08-07 12:36:59,951] Trial 33 finished with value: 0.18444512068632324 and parameters: {'n_estimators': 2000, 'learning_rate': 0.02196598326745786}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 22.8516	valid_1's l1: 44.962
[1000]	training's l1: 19.1954	valid_1's l1: 44.319
[1500]	training's l1: 17.1662	valid_1's l1: 43.7865
[2000]	training's l1: 16.2015	valid_1's l1: 43.6134
Early stopping, best iteration is:
[2328]	training's l1: 15.8169	valid_1's l1: 43.5475
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 21.9087	valid_1's l1: 46.7876
[1000]	training's l1: 18.3217	valid_1's l1: 46.0255
[1500]	training's l1: 16.6185	valid_1's l1: 45.6187
[2000]	training's l1: 15.7122	valid_1's l1: 45.3997
Early stopping, best iteration is:
[2050]	training's l1: 15.6462	valid_1's l1: 45.3807
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 23.3249	valid_1's l1: 47.2336
[1000]	training's l1: 19.796	valid_1's l1: 46.23

[I 2023-08-07 12:37:42,311] Trial 34 finished with value: 0.2096409334368201 and parameters: {'n_estimators': 5000, 'learning_rate': 0.13315368143460193}. Best is trial 15 with value: 0.17323944386353118.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 53.4697	valid_1's l1: 56.5278
[1000]	training's l1: 39.8258	valid_1's l1: 48.9423
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 39.8258	valid_1's l1: 48.9423
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 51.5874	valid_1's l1: 60.5801
[1000]	training's l1: 40.0924	valid_1's l1: 51.998
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 40.0924	valid_1's l1: 51.998
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 54.8189	valid_1's l1: 61.879
[1000]	training's l1: 43.5098	valid_1's l1: 55.0854
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 43.5098	valid_1's l1: 55.0854
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores don

[I 2023-08-07 12:38:01,207] Trial 35 finished with value: 0.17229910321898267 and parameters: {'n_estimators': 1000, 'learning_rate': 0.008452058004905973}. Best is trial 35 with value: 0.17229910321898267.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 29.7008	valid_1's l1: 45.6733
[1000]	training's l1: 24.5338	valid_1's l1: 44.1061
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 24.5338	valid_1's l1: 44.1061
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 28.6623	valid_1's l1: 46.549
[1000]	training's l1: 23.7067	valid_1's l1: 44.9289
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 23.7067	valid_1's l1: 44.9289
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 33.4052	valid_1's l1: 51.3436
[1000]	training's l1: 26.9676	valid_1's l1: 48.7228
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 26.9676	valid_1's l1: 48.7228
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores d

[I 2023-08-07 12:38:18,328] Trial 36 finished with value: 0.18517493259704004 and parameters: {'n_estimators': 1000, 'learning_rate': 0.04671865745887936}. Best is trial 35 with value: 0.17229910321898267.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 53.2369	valid_1's l1: 56.2945
[1000]	training's l1: 39.5219	valid_1's l1: 48.8862
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 39.5219	valid_1's l1: 48.8862
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 51.196	valid_1's l1: 60.9238
[1000]	training's l1: 39.1006	valid_1's l1: 51.7755
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 39.1006	valid_1's l1: 51.7755
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 54.5418	valid_1's l1: 62.2163
[1000]	training's l1: 43.58	valid_1's l1: 55.8652
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 43.58	valid_1's l1: 55.8652
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores don't

[I 2023-08-07 12:38:39,039] Trial 37 finished with value: 0.17040482646838673 and parameters: {'n_estimators': 1000, 'learning_rate': 0.008502109868295124}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 34.0494	valid_1's l1: 46.826
[1000]	training's l1: 27.4985	valid_1's l1: 44.3183
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 27.4985	valid_1's l1: 44.3183
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 33.605	valid_1's l1: 48.5669
[1000]	training's l1: 27.3534	valid_1's l1: 46.5142
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 27.3534	valid_1's l1: 46.5142
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 38.1518	valid_1's l1: 53.2182
[1000]	training's l1: 30.5275	valid_1's l1: 50.2375
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 30.5275	valid_1's l1: 50.2375
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores do

[I 2023-08-07 12:38:58,169] Trial 38 finished with value: 0.173375349863452 and parameters: {'n_estimators': 1000, 'learning_rate': 0.02858784441251459}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 33.9858	valid_1's l1: 46.506
[1000]	training's l1: 27.9346	valid_1's l1: 44.9793
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 27.9346	valid_1's l1: 44.9793
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 33.8806	valid_1's l1: 48.796
[1000]	training's l1: 27.5476	valid_1's l1: 46.7163
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 27.5476	valid_1's l1: 46.7163
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 39.1571	valid_1's l1: 54.171
[1000]	training's l1: 32.0163	valid_1's l1: 51.3135
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 32.0163	valid_1's l1: 51.3135
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores don

[I 2023-08-07 12:39:17,672] Trial 39 finished with value: 0.17961362683657991 and parameters: {'n_estimators': 1000, 'learning_rate': 0.02787900633583838}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 49.3947	valid_1's l1: 53.8051
[1000]	training's l1: 38.4484	valid_1's l1: 48.7888
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 38.4484	valid_1's l1: 48.7888
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 46.8509	valid_1's l1: 56.3553
[1000]	training's l1: 37.0425	valid_1's l1: 49.5511
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 37.0425	valid_1's l1: 49.5511
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 50.9705	valid_1's l1: 59.5463
[1000]	training's l1: 41.7041	valid_1's l1: 54.4757
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 41.7041	valid_1's l1: 54.4757
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores 

[I 2023-08-07 12:39:36,285] Trial 40 finished with value: 0.17698144625793877 and parameters: {'n_estimators': 1000, 'learning_rate': 0.01009159802078098}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 53.1222	valid_1's l1: 56.6689
[1000]	training's l1: 40.3461	valid_1's l1: 49.6426
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 40.3461	valid_1's l1: 49.6426
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 51.7514	valid_1's l1: 60.9915
[1000]	training's l1: 39.7033	valid_1's l1: 51.7672
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 39.7033	valid_1's l1: 51.7672
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 55.1221	valid_1's l1: 62.2887
[1000]	training's l1: 44.0663	valid_1's l1: 55.7966
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 44.0663	valid_1's l1: 55.7966
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores 

[I 2023-08-07 12:39:56,813] Trial 41 finished with value: 0.17301121087773466 and parameters: {'n_estimators': 1000, 'learning_rate': 0.00833914661431023}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 41.6888	valid_1's l1: 50.0637
[1000]	training's l1: 33.2597	valid_1's l1: 46.613
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 33.2597	valid_1's l1: 46.613
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 40.3139	valid_1's l1: 52.1667
[1000]	training's l1: 31.903	valid_1's l1: 47.4916
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 31.903	valid_1's l1: 47.4916
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 44.0742	valid_1's l1: 55.5782
[1000]	training's l1: 35.8803	valid_1's l1: 51.9985
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 35.8803	valid_1's l1: 51.9985
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores don'

[I 2023-08-07 12:40:14,506] Trial 42 finished with value: 0.17684783778727908 and parameters: {'n_estimators': 1000, 'learning_rate': 0.016335250435531783}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 54.4237	valid_1's l1: 57.7746
[1000]	training's l1: 40.5356	valid_1's l1: 49.2759
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 40.5356	valid_1's l1: 49.2759
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 52.5137	valid_1's l1: 62.2081
[1000]	training's l1: 40.0005	valid_1's l1: 52.2714
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 40.0005	valid_1's l1: 52.2714
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 55.9835	valid_1's l1: 62.8975
[1000]	training's l1: 44.0182	valid_1's l1: 55.4717
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 44.0182	valid_1's l1: 55.4717
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores 

[I 2023-08-07 12:40:33,967] Trial 43 finished with value: 0.17520228835388868 and parameters: {'n_estimators': 1000, 'learning_rate': 0.008095229777899502}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 37.3827	valid_1's l1: 48.2625
[1000]	training's l1: 31.0532	valid_1's l1: 46.2071
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 31.0532	valid_1's l1: 46.2071
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 35.5742	valid_1's l1: 49.1033
[1000]	training's l1: 28.8776	valid_1's l1: 46.7934
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 28.8776	valid_1's l1: 46.7934
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 41.5654	valid_1's l1: 54.9211
[1000]	training's l1: 32.9935	valid_1's l1: 50.4517
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 32.9935	valid_1's l1: 50.4517
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores 

[I 2023-08-07 12:40:53,233] Trial 44 finished with value: 0.18622104032418824 and parameters: {'n_estimators': 1000, 'learning_rate': 0.022222160341145304}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 29.1419	valid_1's l1: 45.1436
[1000]	training's l1: 24.2104	valid_1's l1: 43.8336
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 24.2104	valid_1's l1: 43.8336
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 29.8314	valid_1's l1: 47.301
[1000]	training's l1: 24.1527	valid_1's l1: 45.2755
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 24.1527	valid_1's l1: 45.2755
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 33.1377	valid_1's l1: 50.7583
[1000]	training's l1: 27.8204	valid_1's l1: 48.9012
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 27.8204	valid_1's l1: 48.9012
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores d

[I 2023-08-07 12:41:12,150] Trial 45 finished with value: 0.18849646405210346 and parameters: {'n_estimators': 1000, 'learning_rate': 0.04430597825070287}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 29.1816	valid_1's l1: 46.4888
[1000]	training's l1: 22.7383	valid_1's l1: 44.372
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 22.7383	valid_1's l1: 44.372
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 27.5778	valid_1's l1: 47.5201
[1000]	training's l1: 23.2155	valid_1's l1: 46.1529
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 23.2155	valid_1's l1: 46.1529
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 30.9674	valid_1's l1: 50.1627
[1000]	training's l1: 25.1681	valid_1's l1: 47.7226
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 25.1681	valid_1's l1: 47.7226
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores do

[I 2023-08-07 12:41:30,575] Trial 46 finished with value: 0.1891327876513401 and parameters: {'n_estimators': 1000, 'learning_rate': 0.053980978619444904}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 33.8462	valid_1's l1: 46.5212
[1000]	training's l1: 27.9923	valid_1's l1: 44.8159
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 27.9923	valid_1's l1: 44.8159
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 32.2819	valid_1's l1: 47.7079
[1000]	training's l1: 26.2031	valid_1's l1: 45.2944
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 26.2031	valid_1's l1: 45.2944
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 36.3414	valid_1's l1: 52.3738
[1000]	training's l1: 29.0869	valid_1's l1: 48.7849
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 29.0869	valid_1's l1: 48.7849
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores 

[I 2023-08-07 12:41:48,755] Trial 47 finished with value: 0.178483730195554 and parameters: {'n_estimators': 1000, 'learning_rate': 0.03095189838366851}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 37.8258	valid_1's l1: 48.4933
[1000]	training's l1: 29.967	valid_1's l1: 45.3336
[1500]	training's l1: 25.9984	valid_1's l1: 43.7298
[2000]	training's l1: 24.0565	valid_1's l1: 43.3586
[2500]	training's l1: 22.7345	valid_1's l1: 43.111
[3000]	training's l1: 21.7921	valid_1's l1: 42.9149
[3500]	training's l1: 21.0497	valid_1's l1: 42.7653
[4000]	training's l1: 20.4565	valid_1's l1: 42.6293
[4500]	training's l1: 19.9289	valid_1's l1: 42.5094
[5000]	training's l1: 19.4781	valid_1's l1: 42.4385
Early stopping, best iteration is:
[4939]	training's l1: 19.5291	valid_1's l1: 42.4279
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 36.4715	valid_1's l1: 49.6315
[1000]	training's l1: 29.2353	valid_1's l1: 46.4842
[1500]	training's l1: 26.1295	valid_1's l1: 45.6162
[2000]	training's l1: 24.0397	v

[I 2023-08-07 12:43:04,103] Trial 48 finished with value: 0.20030489208917562 and parameters: {'n_estimators': 7000, 'learning_rate': 0.021662304261280282}. Best is trial 37 with value: 0.17040482646838673.


2017-03-01 00:00:00 2017-03-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 59.5938	valid_1's l1: 61.3687
[1000]	training's l1: 42.16	valid_1's l1: 50.3657
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 42.16	valid_1's l1: 50.3657
2017-04-01 00:00:00 2017-04-30 00:00:00 30
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 56.7996	valid_1's l1: 65.7477
[1000]	training's l1: 41.0993	valid_1's l1: 52.3804
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 41.0993	valid_1's l1: 52.3804
2017-05-01 00:00:00 2017-05-31 00:00:00 31
Training until validation scores don't improve for 100 rounds
[500]	training's l1: 60.0851	valid_1's l1: 65.9667
[1000]	training's l1: 46.1936	valid_1's l1: 57.0114
Did not meet early stopping. Best iteration is:
[1000]	training's l1: 46.1936	valid_1's l1: 57.0114
2017-06-01 00:00:00 2017-06-30 00:00:00 30
Training until validation scores don'

[I 2023-08-07 12:43:23,072] Trial 49 finished with value: 0.17298180854435263 and parameters: {'n_estimators': 1000, 'learning_rate': 0.007067306848916712}. Best is trial 37 with value: 0.17040482646838673.


In [ ]:
print(f"\tBest value (rmse): {study.best_value:.5f}")
print(f"\tBest params:")

for key, value in study.best_params.items():
    print(f"\t\t{key}: {value}")
    